fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/ScalarOperatorsReuse.java

```diff
@@ -370,8 +370,7 @@ public Integer visitDictMappingOperator(DictMappingOperator scalarOperator, Void
             return collectCommonOperatorsByDepth(1, scalarOperator);
         }
 
-        // If a scalarOperator contains any non-deterministic function, it cannot be reused
-        // because the non-deterministic function results returned each time are inconsistent.
+
         // a lambda-dependent expressions also can't be reused if reuseLambdaDependentExpr is false.
```

Comment:
We had used other method to distinguish non-deterministic function. So just remove this old comment.